src/filter_mode.rs

```diff
@@ -0,0 +1,60 @@
+use crate::error::Error;
+use std::convert::TryFrom;
+use std::default::Default;
+
+/// Specifies how incoming chat messages should be filtered.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum FilterMode {
+    /// Do not move chat messages out of the Inbox.
+    None,
+
+    /// Let Deltachat move chat messages from Inbox to a configured folder. Requires no server-side
+    /// COI support. This is the default behaviour.
+    Deltachat,
+
+    /// Let the COI server move chat messages to the standard COI/Chats folder.
+    CoiActive,
+
+    /// Let the COI server move chat messages to the standard COI/Chats folder after they have been
+    /// marked as seen.
+    CoiMoveAfterRead,
+}
```

Comment:
We should pick more descriptive names, e.g. using "Client" instead of "Deltachat" and "Server" instead of "Coi".